cyten/tensors.py

```diff
@@ -5087,6 +5087,99 @@ def svd_apply_mask(U: SymmetricTensor, S: DiagonalTensor, Vh: SymmetricTensor, m
     return U, S, Vh
 
 
+def tensor_from_grid(grid: list[list[SymmetricTensor | None]],
+                     labels: Sequence[list[str | None] | None] | list[str | None] | None = None,
+                     dtype: Dtype = Dtype.complex128) -> SymmetricTensor:
+    """Convert a grid of tensors to a single SymmetricTensor.
+
+    The tensors are combined along the first leg in their codomain and the final leg in their
+    domain. All other legs must be identical across all tensors in the grid. For consistency,
+    tensors within the same row must have identical left spaces (first leg in the codomain),
+    and tensors within the same column must have identical right spaces (final leg in the
+    domain).
```

Comment:
I would like a graphical representation sth. like

result  =   sum_{ij}    inclusion_i   grid[i][j]    projection_j

sth like we had on the blackboard as the definition.
We could then write down the result legs in math as direct sums, with the indices i and j.

If you do the sentences, I can do a nice ASCII graphic.